hoodie-utilities/src/main/java/com/uber/hoodie/utilities/HoodieDataImporter.java

```diff
@@ -218,26 +221,32 @@ protected int dataImport(JavaSparkContext jsc) throws IOException {
             properties.put(HoodieTableConfig.HOODIE_TABLE_TYPE_PROP_NAME, cfg.tableType);
 
             HoodieTableMetaClient.initializePathAsHoodieDataset(fs, cfg.targetPath, properties);
+            HoodieWriteClient client = createHoodieClient(jsc, cfg.targetPath, schemaStr, cfg.parallelism);
+
+            SQLContext sqlContext = new SQLContext(jsc);
 
             Path folderPath = new Path(cfg.srcPath);
             RemoteIterator<LocatedFileStatus> locatedFileStatusRemoteIterator = fs
                 .listFiles(folderPath, true);
-            HoodieWriteClient client = createHoodieClient(jsc, cfg.targetPath, schemaStr, cfg.parallelism);
-
-            Job job = Job.getInstance(jsc.hadoopConfiguration());
-            AvroReadSupport.setAvroReadSchema(jsc.hadoopConfiguration(),
-                (new Schema.Parser().parse(schemaStr)));
-            ParquetInputFormat.setReadSupportClass(job, (AvroReadSupport.class));
+            Set<String> paths = new HashSet<String>();
+            while (locatedFileStatusRemoteIterator.hasNext()) {
+                paths.add(locatedFileStatusRemoteIterator.next().getPath().getParent().toString());
+            }
 
-            JavaRDD<GenericRecord> genericRecords = jsc
-                .newAPIHadoopFile(cfg.srcPath, ParquetInputFormat.class, Void.class,
-                    GenericRecord.class, job.getConfiguration()).values();
-            JavaRDD<HoodieRecord<HoodieAvroPayload>> hoodieRecords = genericRecords.map(
-                new Function<GenericRecord, HoodieRecord<HoodieAvroPayload>>() {
+            JavaRDD<HoodieRecord<HoodieAvroPayload>> hoodieRecords = sqlContext.read()
```

Comment:
can we do HiveContext.table() ... this will be to read any table.. Thats where I was going..